lib/linters/single_line_per_property.js

```diff
@@ -1,99 +1,86 @@
 
-var isPreviousValid = function (node, element, index) {
-    var prev = node.get(index - 1);
+module.exports = {
 
-    if (!prev) {
-        // no previous sibling, assume beginning of block without new line, mark invalid
-        return false;
-    } else if (prev.type === 'space' && !prev.toString().match(/\n/)) {
-        // previous sibling not new line
-        return false;
-    } else {
-        return true;
-    }
-};
+    name: 'singleLinePerProperty',
+    nodeTypes: ['block'],
+    message: 'Each property should be on its own line.',
 
-var isNextValid = function (node, element, index) {
-    var next = node.get(index + 2); // We need to move two nodes since the first one is the declaration delimiter (;)
-    var isValid;
+    lint: function singleLinePerPropertyLinter (config, node) {
+        var results = [];
+        var self = this;
 
-    if (!next) {
-        // no next sibling, assume end of block without new line, mark invalid
-        isValid = false;
-    } else {
-        switch (next.type) {
-            case 'space':
-                if (next.content.match(/\n/)) {
-                    isValid = true;
-                } else {
-                    // sibling is not new line
-                    var afterNext = node.get(index + 3);
+        node.forEach('declaration', function (element, index) {
+            var previousValid = self.isPreviousValid(node, element, index);
+            var nextValid = self.isNextValid(node, element, index);
 
-                    if (!afterNext) {
-                        // assume end of block without new line, mark invalid
-                        isValid = false;
-                    } else {
-                        // only valid if followed by comment
-                        if (afterNext.type === 'multilineComment' || afterNext.type === 'singlelineComment') {
-                            isValid = true;
-                        } else {
-                            isValid = false;
-                        }
-                    }
-                }
-                break;
-            case 'singlelineComment': // single-line comment butt up against declaration
-                isValid = true;
-                break;
-            case 'multilineComment': // multi-line comment butt up against declaration
-                isValid = true;
-                break;
-        }
-    }
+            // BOTH previous and next must be valid
+            if (!(previousValid && nextValid)) {
+                results.push({
+                    column: element.start.column,
+                    line: element.start.line,
+                    message: self.message
+                });
+            }
+        });
 
-    return isValid;
-};
+        if (results.length) {
+            return results;
+        }
+    },
 
-module.exports = function (options) {
-    var config = options.config;
-    var node = options.node;
-    var results = [];
+    isPreviousValid: function (node, element, index) {
```

Comment:
I don't think we need to export these functions (`isPreviousValid()` and `isNextValid()`), lets move them outside.